File: src/command.ts
import { cancel, intro, isCancel, multiselect, outro } from "@clack/prompts";
import { defineCommand } from "citty";
import packageJson from "../package.json" with { type: "json" };
import { addService, allServices, type Service } from "./services.js";

const cliName = "komp";

export const main = defineCommand({
  meta: {
    name: cliName,
    version: packageJson.version,
    description: "Add services to your Docker Compose file",
  },
  async run() {
    intro(cliName);
    const res = await multiselect({
      message: "Which service do you want to add?",
      options: allServices.map((s) => ({ value: s, name: s })),
      cursorAt: "postgres",
      required: true,
    });

    if (isCancel(res)) {
      cancel("Operation cancelled.");
      process.exit(0);
    }

    await addService(res as Service[]);
    outro("Done 🐳");
  },
});

File: src/command.rs
use console::style;
use std::{thread, time::Duration};

pub fn run_main() -> std::io::Result<()> {
    let cli_name = String::from("komp");
    cliclack::intro(style(format!("  {}  ", cli_name)).on_yellow().black())?;

    let services_selected = cliclack::multiselect("Which service do you want to add ? 🤔")
        .item("postgres", "PostgreSQL", "")
        .item("mysql", "MySQL", "")
        .item("mailpit", "Mailpit", "")
        .required(true)
        .interact()?;

    let spinner = cliclack::spinner();
    spinner.start("Adding services to your Docker Compose file");
    thread::sleep(Duration::from_secs(1));
    spinner.stop("Services applied 👌");

    cliclack::note("Summary 📜", style(services_selected.join("\n")).yellow())?;

    cliclack::outro("Done 🐳")?;

    Ok(())
}
